utils/backoff.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package utils
+
+import (
+	"time"
+)
+
+const (
+	backoffBase uint64 = 128
```

Comment:
I'm a C programmer myself, but 1/5 I'd do this using the go type system.

Also, FWIW 128ms seems like a pretty long starting timeout.

```go
var backoffTimeouts = []time.Duration {50*time.Millisecond, 100*time.Millisecond, 200*time.Millisecond, 200*time.Millisecond, ...]

// ...
if n > length(backoffTimeouts) {
    // ... error out
}
timeout = backoffTimeouts[n]
```

Reply:
Yeah, 128ms is quite long. But the timeout in `EnsureBot` was previous 1s. So if I were to change the starting timeout to 50ms as you suggesting I would say that I should increase the number of iterations so the max timeout that can occur is atleast 1s.